src/core/src/op/constant.cpp

```diff
@@ -79,6 +79,14 @@ Constant::Constant(const std::shared_ptr<ngraph::runtime::Tensor>& tensor) {
         constructor_validate_and_infer_types();
         allocate_buffer(false);
         tensor->read(get_data_ptr_nc(), tensor->get_size_in_bytes());
+        if (m_element_type == ov::element::string) {
+            // we need to re-initialize memory with separate (newly created) std::string objects with the same values
+            auto size = shape_size(m_shape);
+            auto string_ptr = static_cast<std::string*>(get_data_ptr_nc());
+            std::transform(string_ptr, string_ptr + size, string_ptr, [](std::string value) {
```

Comment:
Here the string is copied twice. First to lambda then as return value.
maybe pass it as reference to lambda and do copy on return ? Will it work?

Reply:
Fixed